python-lib/tagger_formatter.py

```diff
@@ -0,0 +1,443 @@
+"""
+Module to write the output dataframe depending on the output format
+This module inherits from Tagger where the tokenization, sentence splitting and Matcher instanciation has been done
+"""
+from tagger import Tagger
+from collections import defaultdict
+import pandas as pd, numpy as np
+from enum import Enum
+from typing import List, AnyStr, Dict
+from spacy.tokens import Span, Doc
+
+
+class OutputFormat(Enum):
+    ONE_ROW_PER_DOC = "one_row_per_doc"
+    ONE_ROW_PER_DOC_JSON = "one_row_per_doc_json"
+    ONE_ROW_PER_TAG = "one_row_per_tag"
+
+
+class TaggerFormatter:
+    def __init__(self, tagger: Tagger):
+        self.tagger = tagger
+        self.df_duplicated_lines = (
+            pd.DataFrame()
+        )  # useful for mode one row/tag with duplicated documents
+        self.output_df = pd.DataFrame()
+        self.tag_sentences = []
+        self.tag_keywords = []
+        self.matches = False
+        self.output_dataset_columns = self._get_output_dataset_columns()
+
+    def formatting_procedure(self):
+        """
+        Public function called by recipe.py
+        Returns the output dataframe depending on the given output format
+        """
+
+        if self.tagger.mode == OutputFormat.ONE_ROW_PER_TAG.value:
+            return self._row_per_keyword()
+        elif self.tagger.mode == OutputFormat.ONE_ROW_PER_DOC_JSON.value:
+            if self.tagger.category_column:
+                return self._row_per_document_category()
+            else:
+                return self._row_per_document_json()
+        else:
+            if (
+                self.tagger.category_column
+            ):  # one_row_per_doc with category /  one_row_per_doc_json
+                self._row_per_document_category()
+                return self._row_per_document_category_output()
+            else:
+                return self._row_per_document()
+
+    def _row_per_document_json(self):
```

Comment:
Can we try to find a more explicit name for all the pairs of `_row_zzz` and `write_row_zzz`?